### PR TITLE
FEATURE: improve site setting search

### PR DESCRIPTION
**Pull request by SamSaffron**
This improves the site setting search so it performs a somewhat
fuzzy match.

Previously it did not handle seperators such as "space" and a
term such as "min_post_length" would not find "min_first_post_length"

A more liberal search algorithm makes it easier to the AI to
navigate settings.

### CODE REVIEW

File: spec/lib/modules/ai_bot/tools/search_settings_spec.rb [hunk 1]
```diff
@@ -5,13 +5,28 @@
...
   before do
     SiteSetting.ai_bot_enabled = true
     toggle_enabled_bots(bots: [llm_model])
   end
 
-  def search_settings(query)
-    described_class.new({ query: query }, bot_user: bot_user, llm: llm)
+  def search_settings(query, mock: true)
+    search = described_class.new({ query: query }, bot_user: bot_user, llm: llm)
+    search.all_settings = fake_settings if mock
```
Comment (martin-brennan) [inline spec/lib/modules/ai_bot/tools/search_settings_spec.rb:28]: I think you need an `after` block to restore the original value of this, or reset it to nil so the memoization can kick in and re-initialize the settings in [search_settings.rb](https://github.com/discourse/discourse-ai/pull/780/files#diff-2f1ab10bdb1709b638c2b05ff7f7c0940fc62db89332de4c429b76570f294f90)

Comment (martin-brennan) [reply spec/lib/modules/ai_bot/tools/search_settings_spec.rb:28]: Otherwise some test flakiness may occur

Comment (martin-brennan) [reply spec/lib/modules/ai_bot/tools/search_settings_spec.rb:28]: An alternative approach is to do `DiscourseAi::AiBot::Tools::SearchSettings.any_instance.stubs(:site_settings).returns(mock)`, then that will automatically get cleaned up

Comment (martin-brennan) [reply spec/lib/modules/ai_bot/tools/search_settings_spec.rb:28]: Or doing the above with `SiteSetting.all_settings` would avoid needing to add an `all_settings` method in that class entirely :)

**Review by martin-brennan — APPROVED** [review]
Approving with comments about stubbing all settings

File: spec/lib/modules/ai_bot/tools/search_settings_spec.rb (hunk 1 above)
Comment (SamSaffron) [reply spec/lib/modules/ai_bot/tools/search_settings_spec.rb:28]: no need, this is just defining an instance :) the instance and the override vanish after the test

Comment (martin-brennan) [reply spec/lib/modules/ai_bot/tools/search_settings_spec.rb:28]: Oh wow duh :facepalm:

**Pull request merged**